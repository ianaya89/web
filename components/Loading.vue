<template>
  <div v-if="loading" class="loading-page">
    <logo-loader class="m-auto w-16" />
  </div>
</template>
<script>
import LogoLoader from '@/components/LogoLoader'

export default {
  name: 'Loading',

  components: {
    LogoLoader
  },

  data: () => ({
    loading: false
  }),

  methods: {
    start() {
      this.loading = true
    },
    finish() {
      this.loading = false
    }
  }
}
</script>

<style lang="postcss" scoped>
.loading-page {
  @apply .fixed .top-0 .left-0 .w-full .h-full .p-96 .text-center;
  background: rgba(255, 255, 255, 0.8);
  z-index: 9999;
}
</style>
