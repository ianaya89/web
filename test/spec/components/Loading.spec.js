import { shallowMount } from '@vue/test-utils'
import Loading from '@/components/Loading'

describe('Loading', () => {
  it('renders default loading', () => {
    const wrapper = shallowMount(Loading)
    expect(wrapper.element).toMatchSnapshot()

    wrapper.vm.start()
    expect(wrapper.element).toMatchSnapshot()
    wrapper.vm.finish()
    expect(wrapper.element).toMatchSnapshot()
  })
})
