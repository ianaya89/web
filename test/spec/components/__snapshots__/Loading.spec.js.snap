// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders default loading 1`] = `<!---->`;

exports[`Loading renders default loading 2`] = `<!---->`;

exports[`Loading renders default loading 3`] = `<!---->`;
